src/rm.js

```diff
@@ -48,7 +48,25 @@ function rmdirSyncRecursive(dir, force) {
 
   var result;
   try {
-    result = fs.rmdirSync(dir);
+    // Retry on windows, sometimes it takes a little time before all the files in the directory are gone
+    var start = Date.now();
+    while (true) {
+      try {
+        result = fs.rmdirSync(dir);
+        if (fs.existsSync(dir)) throw { code: "EEXIST" }
```

Comment:
@andreialecu - I _completely_ missed that outer `catch`. Thanks for being patient with me, and pointing that out. That solves 1 and 3, and I'd just replace 2 with `EAGAIN`, because trying again _may_ succeed, whereas `EEXIST` is meant to indicate that creating a new file failed because one already exists, and the user _should not_ try again until that's resolved.

Does that make sense?